dali/python/nvidia/dali/external_source.py

```diff
@@ -251,10 +277,14 @@ class ExternalSource():
     For the GPU, to avoid extra copy, the provided buffer must be contiguous. If you provide a list
     of separate Tensors, there will be an additional copy made internally, consuming both memory
     and bandwidth.
+
+`batch` : optional
+    If set to ``True`` or ``None``, the ``source`` is expected to produce an entire batch at once.
+    If set to ``False``, the ``source`` is called per-sample. Its first parameter is sample index.
```

Comment:
How would you know the batch size in this case?

Reply:
The whole point is to not have to know it - you just produce consecutive samples and that's it.